Source/Audio/Audio/AudioManager.cpp

```diff
@@ -0,0 +1,130 @@
+#include "AudioManager.h"
+
+AudioManager::AudioManager()
+{
+
+}
+
+bool AudioManager::Init()
+{
+    if (_audioEngine != nullptr)
+        return false;
+
+    ma_engine* engine = new ma_engine();
+    ma_result result = ma_engine_init(NULL, engine);
+
+    if (result != MA_SUCCESS)
+    {
+        delete engine;
+        return false;
+    }
+
+    _audioEngine = engine;
+    return true;
+}
+
+void AudioManager::Cleanup()
+{
+    if (_audioEngine == nullptr)
+        return;
+
+    ma_engine_uninit(_audioEngine);
+
+    delete _audioEngine;
+    _audioEngine = nullptr;
+}
+
+void AudioManager::PlaySoundFile(f32 volume, bool isLooping, bool endAllOtherAudio)
+{
+    if (_audioEngine == nullptr)
+        return;
+
+    if (endAllOtherAudio && ma_sound_is_playing(_assetBrowserSound))
+        ma_sound_uninit(_assetBrowserSound);
+
+    _assetBrowserSound = new ma_sound();
```

Comment:
We don't perticularly like having implementation details like "asset browser sound" in the AudioManager. It would be better if you could create channels in the AudoManager, and then you can play sounds in a specific channel, stop everything playing in a specific channel etc. Then for example the AudioDebugWindow or whatever you would call the editor window would create a channel for itself and use that channel to play/stop sounds.

Reply:
Same with _consoleSounds

Reply:
When I say Channel here, I mean MA's Sound Groups, like "Music", "Ambience", "Sound effects" etc. Not for example left and right channel on headphones. :P